src/cli.ts

```diff
@@ -245,21 +246,42 @@ program
     `List all locations to run the synthetics monitors. Pass optional '--url' and '--auth' to list private locations.`
   )
   .option('--url <url>', 'Kibana URL to fetch all public and private locations')
-  .option(
-    '--auth <auth>',
-    'API key used for Kibana authentication(https://www.elastic.co/guide/en/kibana/master/api-keys.html).'
+  .addOption(
+    new Option(
+      '--auth <auth>',
+      'API key used for Kibana authentication(https://www.elastic.co/guide/en/kibana/master/api-keys.html).'
+    ).env('SYNTHETICS_API_KEY')
   )
   .action(async (cmdOpts: LocationCmdOptions) => {
+    const workDir = cwd();
+    let url = cmdOpts.url;
+    let tearDown = () => {};
     try {
-      let locations = Object.keys(LocationsMap);
-      if (cmdOpts.auth && cmdOpts.url) {
-        const allLocations = await getLocations(cmdOpts);
-        locations = formatLocations(allLocations);
+      if (!url) {
+        tearDown = await globalSetup({ inline: false, ...program.opts() }, [
```

Comment:
Move this out of the try block, not required. we don't need the full setup since setup also involves loading test files which are not required. You can just do transform. 
```ts
 const revert = installTransform();
 finally {
 revert()
}

```